src/python/pants/backend/codegen/avro/java/rules_integration_test.py

```diff
@@ -172,3 +223,10 @@ def assert_gen(addr: Address, expected: Iterable[str]) -> None:
         Address("src/avro/dir1", relative_file_path="user.avsc"),
         ["src/avro/org/pantsbuild/contrib/avro/User.java"],
     )
+
+    ct = expect_single_expanded_coarsened_target(rule_runner, Address("src/jvm"))
+    check_results = rule_runner.request(
+        CheckResults,
+        [JavacCheckRequest([JavacCheckRequest.field_set_type.create(ct.representative)])],
```

Comment:
In future, `RenderedClasspath` might allow for a simpler/more-thorough test that something compiles: https://github.com/pantsbuild/pants/blob/900fe9287e76e963cac1844453792746053cb83d/src/python/pants/jvm/compile_test.py#L291-L320